workers/unity/Packages/io.improbable.worker.sdk/ParameterConversion.cs

```diff
@@ -0,0 +1,140 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.ConstrainedExecution;
+using System.Runtime.InteropServices;
+
+namespace Improbable.Worker.CInterop.Internal
+{
+    // Merge with /worker_sdk/csharp_cinterop/sdk/worker/internal/ParameterConversion.cs
+    // when changes upstreamed to Worker SDK
+    internal static class ParameterConversion
+    {
+        internal class WrappedGcHandle : CriticalFinalizerObject, IDisposable
+        {
+            private GCHandle handle;
+
+            public WrappedGcHandle(object obj)
+            {
+                handle = GCHandle.Alloc(obj);
+            }
+
+            ~WrappedGcHandle()
+            {
+                if (handle.IsAllocated)
+                {
+                    handle.Free();
+                }
+            }
+
+            public void Dispose()
+            {
+                if (handle.IsAllocated)
+                {
+                    handle.Free();
+                }
+
+                GC.SuppressFinalize(this);
+            }
+
+            public IntPtr Get()
+            {
+                return GCHandle.ToIntPtr(handle);
+            }
+        }
+
+        public static unsafe CEventTrace.SpanId ConvertSpanId(SpanId spanId)
+        {
+            var internalSpanId = new CEventTrace.SpanId();
+            ApiInterop.Memcpy(internalSpanId.Data, spanId.Data, (UIntPtr) 16);
+            return internalSpanId;
+        }
+
+        public static unsafe CEventTrace.Event ConvertEvent(Event eventToConvert)
+        {
+            var internalEvent = new CEventTrace.Event();
+            internalEvent.UnixTimestampMillis = eventToConvert.UnixTimestampMillis;
+            internalEvent.Id = ConvertSpanId(eventToConvert.Id);
+            internalEvent.Data = eventToConvert.Data?.eventData.AddrOfPinnedObject() ?? IntPtr.Zero;
```

Comment:
Same comment here, don't we want the handle underlying `eventData` rather than a pointer to the wrapper class?